Assets/MixedRealityToolkit/Extensions/BoundsExtensions.cs

```diff
@@ -354,54 +354,62 @@ public static void GetColliderBoundsPoints(GameObject target, List<Vector3> boun
             Collider[] colliders = target.GetComponentsInChildren<Collider>();
             for (int i = 0; i < colliders.Length; i++)
             {
-                if (ignoreLayers == (1 << colliders[i].gameObject.layer | ignoreLayers))
-                {
-                    continue;
-                }
+                GetColliderBoundsPoints(colliders[i], boundsPoints, ignoreLayers);
+            }
+        }
 
-                if(colliders[i] is SphereCollider)
-                {
-                    SphereCollider sc = colliders[i] as SphereCollider;
-                    Bounds sphereBounds = new Bounds(sc.center, Vector3.one * sc.radius * 2);
-                    sphereBounds.GetFacePositions(sc.transform, ref corners);
-                    boundsPoints.AddRange(corners);
-                }
-                else if(colliders[i] is BoxCollider)
-                {
-                    BoxCollider bc = colliders[i] as BoxCollider;
-                    Bounds boxBounds = new Bounds(bc.center, bc.size);
-                    boxBounds.GetCornerPositions(bc.transform, ref corners);
-                    boundsPoints.AddRange(corners);
+        /// <summary>
+        /// Method to get bounds from a single Collider
+        /// </summary>
+        /// <param name="collider">Target collider</param>
+        /// <param name="boundsPoints">array reference that gets filled with points</param>
+        /// <param name="ignoreLayers">layerMask to simplify search</param>
+        public static void GetColliderBoundsPoints(Collider collider, List<Vector3> boundsPoints, LayerMask ignoreLayers)
```

Comment:
This looks like a function that would be a great candidate for a unit test / edit mode test, since it has known inputs and outputs. It will be great to have to not only verify your changes but more importantly make sure nobody breaks you in the future :-)

Reply:
Since this PR fixes existing functionality I would be okay to complete this PR without tests, but would strongly suggest to add if possible. If not, could you please file an issue to write a test for this function so we don't forget? You can either assign it to yourself if you have time to help, or we can find somebody later to help with this.

Reply:
Sadly, I don't have the time to write tests too. Writing fixes is already a time consuming process. Btw. I'm not allowed or don't know how to assign issues to myself.

> could you please file an issue to write a test for this function so we don't forget

It's not new really. I only extracted this from the other existing method which was meant for collider arrays only and I needed it for one collider at a time. The other method hopefully does have a test already.

Reply:
Filed #5486